projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -66,6 +76,21 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
     }
   }
 
+  private handleFormArrayControls(obj: any) {
+    Object.entries(obj).forEach(([key, value]) => {
+      if (this.formGroup.get(key) instanceof FormArray && Array.isArray(value)) {
+        const formArray: FormArray = this.formGroup.get(key) as FormArray;
+        // we need to empty the array before adding the new value
+        // otherwise after selecting another preset, if the number of values
+        // is different it will error
```

Comment:
While simple, I suspect there may be some edge case bugs this approach may cause, given form controls manage some internal state. I think the more verbose option of setting the value for each existing control and adding to or removing from the end of the array where necessary may be the best option. Just thinking about some things like async validators refiring for the same value etc

Reply:
Do you have any concrete example in mind? Because I'm not really sure to follow here.
If you have a real edge case to demo it might help to understand :)

Reply:
Summarising IRL conversation; I think there is two options - one is to add a method that implementing classes must implement if they have a `FormArray`, and the other is to just have the implementing class manage the `FormArray` independently. Both options are very valid, so we should probably implement the first generically, and demo _both_.


```ts
// in this class

protected getNewFormControlForFormArray(key: WellTypedKey): AbstractControl {
  throw new Error(`You must implement this method for a form array with name ${key}`);
}


// in SpeederComponent
protected getNewFormControlForFormArray(key: WellTypedKey): AbstractControl {
  switch(key) {
   case 'colors':
    return new FormControl();
  }
}

// in Droid compontn

public colors = new FormArray();
protected transformToFormGroup({colors, ...restOfDroid}) {
   return restOfDroid
}

protected transformFromFormGroup(fg){
   return {...fg, colors: this.colors.value};
}

```